e2e/kube.go

```diff
@@ -76,32 +76,29 @@ func getClusterKubeconfigBytes(ctx context.Context, resourceGroupName, clusterNa
 }
 
 // this is a bit ugly, but we don't want to execute this piece concurrently with other tests
-func ensureDebugDaemonset(ctx context.Context, kube *Kubeclient) error {
-	manifest := getDebugDaemonset()
-	var ds v1.DaemonSet
-
-	if err := yaml.Unmarshal([]byte(manifest), &ds); err != nil {
-		return fmt.Errorf("failed to unmarshal debug daemonset manifest: %w", err)
+func ensureDebugDaemonsets(ctx context.Context, kube *Kubeclient) error {
+	manifests := getDebugDaemonsetManifests()
+	for _, manifest := range manifests {
+		if err := createDebugDeployment(ctx, kube, manifest); err != nil {
+			return err
+		}
 	}
 
-	desired := ds.DeepCopy()
-	_, err := controllerutil.CreateOrUpdate(ctx, kube.Dynamic, &ds, func() error {
-		ds = *desired
-		return nil
-	})
+	return nil
+}
 
-	if err != nil {
-		return fmt.Errorf("failed to apply debug daemonset: %w", err)
+func getDebugDaemonsetManifests() []string {
+	return []string{
+		getDebugDaemonsetTemplate(hostNetworkDebugPodNamePrefix, "nodepool1", true),
+		getDebugDaemonsetTemplate(nonHostNetworkDebugPodNamePrefix, "nodepool2", false),
```

Comment:
im confused, with the new validation, don't we want to schedule a new pod (e.g. just a Pod object, not Deployment/Daemonset) with a node selector selecting for the abe2e node we're testing (by unique name/label), that runs outside the host network?

with the current implementation we're basically creating 2 separate deployments (also not sure the difference between nodepool1 and nodepool2, is nodepool2 the label the e2e nodes will have? don't we want to have a _unique_ selector for each abe2e node?)

Reply:
was your thinking that if all abe2e nodes have the `nodepool2` label then this would automatically schedule onto all abe2e nodes whenever they come up (though it would need to be a DS for that to work)?

Reply:
This "deployment" is actually a daemonset if you look at the ab e2e cluster resources which are deployed. 

And yeah basically the thinking is that this will just create a separate pod for each of the test nodes and run required checks. The naming there was a little bit confusing between daemonset/deployment, i'm guessing what gets deployed in the end is the type we marshall to, hence we end up with a daemonset.

Reply:
`Nodepool2` is the label that all the e2e test cases nodes have, that's correct. It was the easiest/quickest way for me to get them to deploy only on the test case nodes, so I ran with it.

Reply:
okay, can you please also change the `Kind` in the manifest to Daemonset so we avoid all this confusion lol